README.rst

```diff
@@ -1,6 +1,220 @@
+                                           title='Todo #1'))
+
+    # Get all records
+    record = collection.get_all_records()
+    record = collection.get_record(id='89881454-e4e9-4ef0-99a9-404d95900352')
+    collection.save_record(record)
+    collection.save_records([record1, record2])
+    collection.delete_record(id='89881454-e4e9-4ef0-99a9-404d95900352')
+    collection.delete_records([record1, record2])
+
+
+Handling conflicts
+------------------
+
+XXX
+
+Permissions
+===========
+
+ By default, the authenticated user will get read and write access to the
+ manipulated objects. It is possible to change this behavior by passing a dict
+ to the `permissions` parameter.
+
+ .. code-block:: python
+
+    record = collection.create_record(
+        data={},
+        permissions={'read': ['group:groupid']})
+
+.. note::
+
+    Every creation or modification operation on a distant object can be given
+    a `permissions` parameter.
+
+The `Bucket`, `Collection`, `Group` and `Record` class have a special
+`permissions` object that can be mutated in order to update the permissions
+model attached to the object.
+
+.. code-block:: python
+
+    bucket = Bucket('personal', auth=('alexis', 'p4ssw0rd'))
+
+    # XXX We need to find a way to get other's names from kinto, this isn't
+    # realistic.
+    friends = ['natim', 'niko', 'mat', 'tarek']
+    bucket.permissions.write += friends
+    bucket.permissions.create_collection += friends
+
+    # You *need* to call save in order to have these changes reflected in the
+    # remote.
+    bucket.permissions.save()
+
+Groups
+======
+
+Giving specific permissions to specific users can be handy sometimes, but
+quickly becomes a pain to maintain if many permissions needs to be given to
+different set of people.
+
+In order to handle this better, Kinto has a concept of groups. Groups represent
+a set of individuals, described by a name. Individuals can then be added and
+removed from the group, and permissions can be given to the group rather than
+the individuals.
+
+.. note::
+
+    Groups are attached to a bucket (and not to a collection). As such they
+    can be shared accross different collections of the same bucket.
+
+Groups can be manipulated like python sets.
+
+.. code-block:: python
+
+    group = bucket.create_group('moderators', ['list', 'of', 'users'])
+    group.add('niko')
+    group.remove('remy')
+    group.clear()  # Remove everyone in the group (except yourself)
+    group.save() # XXX Add an option to remove current user from the group?
+
+
+Sending requests in batch
+=========================
+
+Sometimes, it is useful to issue multiple operations in batch, to avoid
+sending many requests to the same server. This is an especially useful thing
+when operations have been done offline and the server needs a refresh.
+
+Batch operations can be done using a python context manager (the `with`
+statement).
+
+.. code-block:: python
+
+    with kintoclient.batch() as session:
```

Comment:
It is hard to see the benefit of using session in this case rather than doing directly:

```
todo = session.get_collection('todo', bucket='personal')
# Pile up your operations here.
todo.save_records(records)
```

Isn't it save_records doing a batch request already?